Doc/tools/extensions/pyspecific.py

```diff
@@ -393,7 +394,23 @@ def run(self):
         return PyMethod.run(self)
 
 
-# Support for documenting version of removal in deprecations
+# Support for documenting version of changes, additions, deprecations
+
+def expand_version_arg(argument, env):
+    """Expand "next" to the current version"""
+    if argument == 'next':
+        return sphinx_gettext('{} (unreleased)').format(env.config.release)
+    return argument
+
+
+class PyVersionChange(sphinx.directives.other.VersionChange):
+    def run(self):
+        env = self.state.document.settings.env
+        self.arguments = (
+            expand_version_arg(self.arguments[0], env),
+            *self.arguments[1:])
+        return super().run()
+
```

Comment:
@AA-Turner, Does this approach look OK to you?

I want to turn `.. versionadded:: next` into `.. versionadded:: 3.14.15`, and do it by subclassing `sphinx.directives.other.VersionChange`, replacing `self.arguments`, and using `app.add_directive(..., override=True)`.

Reply:
Looks great! I've added a few comments.

Reply:
Oh, actually -- ``sphinx.directives.other`` is a compat import. This should be ``sphinx.domains.changeset`` (since Sphinx 1.8).

Reply:
https://www.sphinx-doc.org/en/master/extdev/deprecated.html suggests using `sphinx.directives.other`. Should it be updated?

Reply:
> [sphinx-doc.org/en/master/extdev/deprecated.html](https://www.sphinx-doc.org/en/master/extdev/deprecated.html) suggests using `sphinx.directives.other`. Should it be updated?

Ah, this must have been missed whenever ``VersionChange`` moved to the changeset domain. Will update.

A